### ISSUE TITLE
Scala3doc: No DRI found for query

### ISSUE THREAD

**Issue by ekrich** [lampepfl/dotty#14212]
## Compiler version

`3.1.0`

## Minimized code
The `trait Config` has docs that point to `ConfigValue`.

```Scala
* <p>
 * Convert a <code>Config</code> to a JSON or HOCON string by calling
 * [[#root root]] to get the [[ConfigObject]] and then call
 * [[ConfigValue!.render:String* render]]
 * on the root object, <code>myConfig.root.render</code>. There's also a variant
 * [[ConfigValue!.render(options:org\.ekrich\.config\.ConfigRenderOptions)* render(ConfigRenderOptions)]]
 * inherited from [[ConfigValue]] which allows you to control
 * the format of the rendered string. (See {@link ConfigRenderOptions}.) Note
 * that <code>Config</code> does not remember the formatting of the original
 * file, so if you load, modify, and re-save a config file, it will be
 * substantially reformatted.
```

## Output

[info] Main Scala API documentation to /Users/eric/workspace/sconfig/sconfig/jvm/target/scala-3.1.0/api...
[info] Skipping unused scalacOptions: -deprecation, -feature
[warn] -- Warning: sconfig/shared/src/main/scala/org/ekrich/config/Config.scala:171:6 -
[warn] 171 |trait Config extends ConfigMergeable {
[warn]     |      ^
[warn]     |      No DRI found for query: ConfigValue!.render:String*
[warn] one warning found
[info] Main Scala API documentation successful.

## Expectation

Works as it did in older versions. See https://www.javadoc.io/doc/org.ekrich/sconfig_2.11/latest/index.html#org.ekrich.config.Config

**Comment by pikinier20** [comment]
This problem is related to: #11119
We don't support method overloading in our member lookup like the old Scaladoc does. This bug shows that our query parser is not 100% compliant with the old parser. I think this problem should be resolved together with method overloading support.

**Comment by abgruszecki** [comment]
@pikinier20 It wasn't really a goal to be 100% compliant with the old parser. If you take a look at https://dotty.epfl.ch/docs/usage/scaladoc/linking.html, the extensions mentioned there are technically incompatible with the old parser. Reg. overloading resolution, I more or less made a conscious choice not to support it the way Scaladoc2 supported it (based on regexes), because it seemed like a really hacky feature that wasn't even used in the community build.

**Comment by ekrich** [comment]
When I ported this from Java to Scala it was really hard to make the links work similar - really a nightmare. I am excited to see something better and easier for authors.

Now moving forward I could change the format and only publish Scala 3 docs but there needs to be a plan to move forward as many library authors have cross builds. For now I have it disabled so I can publish my project - others have done the same.

**Comment by bblfish** [comment]
could the meaning of DRI be written out in full? Looking at the error code in sbt I really had no clue where it came from.

**Comment by abgruszecki** [comment]
@bblfish DRI is internal Scala3doc terminology. I agree we should avoid it in the error message, it's certainly not helpful. We could just say that we failed to resolve a query.

**Comment by bblfish** [comment]
> @bblfish DRI is internal Scala3doc terminology. I agree we should avoid it in the error message, it's certainly not helpful. We could just say that we failed to resolve a query.

Thanks :-)
How do I go about resolving that error message?

**Comment by abgruszecki** [comment]
In your specific case, this one : ` run.cosy.http.headers.Rfc5234` ? If that's a valid identifier, then not looking it up is 100% a bug, please open an issue and someone will take a look at fixing it.

**Comment by ekrich** [comment]
I am not really seeing the Scaladoc docs here: https://dotty.epfl.ch/
These aren't the greatest and I had to search high and low for tips. https://docs.scala-lang.org/style/scaladoc.html

Has there been any discussion about links to overloads? I would like to get rid of the warning but then the doc links don't really point to the overloaded method so it is not really correct.

I found I can get rid of the warning by changing from
`[[ConfigValue!.render:String render]]` to
`[[ConfigValue!.render():String render]]` or
`[[ConfigValue!.render render]]` but then the links still point to
`[[ConfigValue!.render(options:org\.ekrich\.config\.ConfigRenderOptions)* render(ConfigRenderOptions)]]`

It would be nice if this was a bit easier.

**Comment by pikinier20** [comment]
@ekrich The problem with overloads is that it's not supported by the query parser which was rewritten in Scaladoc 3. So, we are always picking the same method ignoring the overload. See: https://github.com/lampepfl/dotty/issues/11119

**Comment by jam01** [comment]
Is there a way to suppress the warnings?

**Comment by ekrich** [comment]
I get this error in `2.13` that hints at how to suppress it.
```
[warn] Applicable -Wconf / @nowarn filters for this warning: msg=<part of the message>, cat=scaladoc, site=org.ekrich.config.Config
```

**Comment by mr-git** [comment]
Having project with `scalaVersion := "3.6.3"`

Using `ResourceApp.Forever` from Cats Effect with code like:
```scala
object Main extends ResourceApp.Forever:
  override def run(args: List[String]): Resource[IO, Unit] = {
// ...
```

Got warning:
```text
[warn] 9 |  override def run(args: List[String]): Resource[IO, Unit] = {
[warn]   |               ^
[warn]   |               No DRI found for query: ResourceApp.run
```

For me it was **impossible** to understand what it complains about! 

After finding this issue I understood that there is some problem with ScalaDoc, but it left me puzzled: what is wrong with lacking ScalaDoc for overridden function?

Adding minimal ScalaDoc as follows "fixes" the warning:
```scala
  /**  */
  override def run(args: List[String]): Resource[IO, Unit] = {
```
Basically it asks to provide `at least some` ScalaDoc explicitly for overridden function 🤷 

This is very bad warning message!

**Comment by ekrich** [comment]
Is there any possibility this could be fixed?

**Comment by mr-git** [comment]
@som-snytt, you have been improving messages in many cryptic errors, could you please tackle this one?

Should we create dedicated issue for incomplete error message?

**Comment by Gedochao** [comment]
> Is there any possibility this could be fixed?

I'll see what we can do.
Don't expect it earlier than 3.8.1, however.

**Comment by SolalPirelli** [comment]
The error messages no longer mention "DRI"s. Closing, but if anyone has a case where something can't be resolved by Scaladoc when you think it should be, do open an issue.
